### PR TITLE
Initial code cleanup / refactor.

### PR DESCRIPTION
**Pull request by k15z**
This renames some of the key variables/methods to comply with the design document. It also moves the data-driven / "end-to-end" tests into a separate directory.

### CODE REVIEW

File: sdmetrics/detection/tabular/base.py [hunk 1]
```diff
@@ -35,12 +35,27 @@ def predict_proba(self, X):
...
+            real_tables (dict): A dictionary mapping table names to dataframes.
+            synthetic_tables (dict): A dictionary mapping table names to dataframes.
+
+        Yields:
+            Metric: The next metric.
+        """
+        yield from self._single_table(metadata, real_tables, synthetic_tables)
+        yield from self._parent_child(metadata, real_tables, synthetic_tables)
+
+    def _single_table(self, metadata, real_tables, synthetic_tables):
```
Comment (kveerama) [inline sdmetrics/detection/tabular/base.py]: name it _single_table_detection perhaps?

File: sdmetrics/detection/tabular/base.py [hunk 2]
```diff
@@ -54,22 +69,24 @@ def metrics(self, metadata, real_tables, fake_tables):
                 domain=(0.0, 1.0)
             )
 
-            # Parent-Child Table Detection
+    def _parent_child(self, metadata, real_tables, synthetic_tables):
```
Comment (kveerama) [inline sdmetrics/detection/tabular/base.py]: maybe name it  _combined_table_detection

File: sdmetrics/statistical/bivariate/base.py [hunk 3]
```diff
@@ -15,15 +15,15 @@ class BivariateMetric():
     dtypes = []
 
     @staticmethod
-    def metric(real, fake):
+    def metric(real_2d, synthetic_2d):
         """This function is expected to perform a statistical test on the two
         samples, each of which is two-dimensional, and return a tuple containing
```
Comment (kveerama) [inline sdmetrics/statistical/bivariate/base.py]: change to "...to perform a statistical test on two columns.."

File: sdmetrics/statistical/univariate/base.py [hunk 4]
```diff
@@ -29,19 +29,31 @@ def metric(real_column, fake_column):
...
-        for name in tables:
-            dtypes = metadata.get_dtypes(name)
-            real = real_tables[name]
-            fake = synthetic_tables[name]
-            yield from self._handle(name, dtypes, real, fake)
-
-    def _handle(self, name, dtypes, real, fake):
+        """
+        This function iterates over all the columns in all the tables and, if the
+        data types match, it yields the Metric.
```
Comment (kveerama) [inline sdmetrics/statistical/univariate/base.py]: Perhaps change to "This function iterates over all the columns in all the tables and, if the data type of a column matches the data type for which this metric is defined, it computes the metric for that column and yields it."

File: sdmetrics/statistical/utils.py [hunk 5]
```diff
@@ -2,27 +2,27 @@
 from collections import Counter
 
 
-def frequencies(real, fake):
+def frequencies(real, synthetic):
     """
     Given two iterators containing categorical data, this transforms it into
     observed/expected frequencies which can be used for statistical tests. It
-    adds a regularization term to handle cases where the fake data contains
+    adds a regularization term to handle cases where the synthetic data contains
```
Comment (kveerama) [inline sdmetrics/statistical/utils.py]: is "regularization" the right word here?

File: sdmetrics/statistical/univariate/base.py (hunk 4 above)
Comment (k15z) [reply sdmetrics/statistical/univariate/base.py]: Done.

File: sdmetrics/statistical/utils.py (hunk 5 above)
Comment (k15z) [reply sdmetrics/statistical/utils.py]: Additive smoothing / Laplace smoothing / pseudo-counts is often treated as regularization so I think it's fine here.

File: sdmetrics/statistical/bivariate/base.py (hunk 3 above)
Comment (k15z) [reply sdmetrics/statistical/bivariate/base.py]: Changed to "This function is expected to perform a statistical test on the two samples, each of which contains two columns, and..."

File: sdmetrics/detection/tabular/base.py (hunk 1 above)
Comment (k15z) [reply sdmetrics/detection/tabular/base.py]: Done.

File: sdmetrics/detection/tabular/base.py (hunk 2 above)
Comment (k15z) [reply sdmetrics/detection/tabular/base.py]: How about `_parent_child_detection`? "combined table" is a bit unclear/vague.

**Pull request merged**